### PR TITLE
Haoyue fix role title

### PR DESCRIPTION
**Pull request by oliviahyw**
# Description

Screenshot 2024-09-20 at 1 22 32 PM

## Related PRS (if any):
To test this backend PR you need to checkout the [#2700](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2700) frontend PR.
…

## Main changes explained:
- Added the validation for Title Code field.
…

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. follow the steps in the front end PR.

## Screenshots or videos of changes:

https://github.com/user-attachments/assets/4ffbc7b3-7e89-4f90-95f4-738412066186

## Note:
Include the information the reviewers need to know.

### CODE REVIEW

**Review by ArtemisNyx3 — APPROVED** [review]
Approved frontend pr
changes working upto spec

**Review by Logeshwari-Renu — APPROVED** [review]
Tested the PR and the functionality works fine. Left the review remarks in [Frontend](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2700#pullrequestreview-2319857342)

**Review by chu-ziyu — APPROVED** [review]
I have tested this PR and it worked as expected. Details comments left in frontend #2700.

**Review by AshishNags — DISMISSED** [review]
Tested this PR, every functionality is working as expected except for one small thing, Title Code is taking first 5 characters from Title Name including the spaces and is giving the error popup, I suggest trimming the white spaces before using it for the Title Code..

Screenshot 2024-10-09 131516

File: src/controllers/titleController.js [hunk 1]
```diff
@@ -10,193 +10,193 @@ const getAllTeamCodeHelper = controller.getAllTeamCodeHelper;
...
+      return;
+    }
+
+    // validate if team exist
+    if (title.teamAssiged && title.teamAssiged._id === 'N/A') {
+      res.status(400).send({ message: 'Team not exists.' });
+      return;
+    }
+
+    title
```
Comment (Ankuriboh) [inline src/controllers/titleController.js:79]: I would suggest utilizing [mongoose's built-in schema validation](https://mongoosejs.com/docs/validation.html). Then you wouldn't need to use multiple `if`s to manually validate the data.

**Review by Ankuriboh — APPROVED** [review]
Can confirm that code works on my end. For further improvements, we should utilize `mongoose`'s built-in schema validator. And if better, use some middleware to automatically validate incoming data bodies. But this is a story for another day and another issue.

**Comment by Anoushka21** [comment]
Tested out frontend- working according to spec.

**Review by Anoushka21 — APPROVED** [review]

**Review by hqzhu0913 — COMMENTED** [review]
Hi,
I found the UI is not the same as in the video. But the function work as explained. The code can only contain alphabetic and up to 5 characters. The rest of images are in the front end review.

**Review by zhiminV — APPROVED** [review]
I have check out this pr in backend and checked that the frontend work as expected in both light and dark mode
Screenshot 2024-10-26 at 3 07 02 PM

**Review by varun-elango — APPROVED** [review]
The functionality is working as expected with the corresponding branch on the frontend.
PR2700+1111_2

**Comment by one-community** [comment]
Thank you all, merging!

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**